crc32.go

```diff
@@ -4,13 +4,88 @@ const (
 	crc32Polynomial = uint32(0xffffffff)
 )
 
-// computeCRC32 computes a CRC32
-// https://stackoverflow.com/questions/35034042/how-to-calculate-crc32-in-psi-si-packet
+// this solution based on vlc implementation + static crc table (1kb additional memory on start, without reallocations)
+// table generated with following function
+/*
+func GenerateTables() {
+	var i uint32
+	fmt.Print(`var tableCRC32 = [256]uint32{`)
+	for i = 0; i < 256; i++ {
+		var k uint32
+		for j := (i << 24) | 0x800000; j != 0x80000000; j <<= 1 {
+			if (k^j)&0x80000000 != 0 {
+				k = (k << 1) ^ 0x04c11db7
+			} else {
+				k = (k << 1) ^ 0
+			}
+		}
+		if i != 0 {
+			fmt.Print(`, `)
+		}
+		if i%8 == 0 {
+			fmt.Print("\n")
+		}
+		fmt.Printf("%#X", k)
+	}
+	fmt.Println(`}`)
+}
+
+*/
+// https://github.com/videolan/vlc/blob/master/modules/mux/mpeg/ps.c
+
+var tableCRC32 = [256]uint32{
```

Comment:
Could you create a command to generate this table in a separate go file?

1) Add an `internal/cmd/crc32_table` folder
2) Add a `main.go` in it that would create a `crc32_table.go` with only the `tableCRC32` variable in it.
3) Also make sure to add `// Code generated by astits. DO NOT EDIT.` a the top of this file before the package name

That way we have a command to generate the file

Reply:
All done! This is my first attempt with code generation, so it may be kind of ugly